src/test/java/com/mael/payroll/paymentTypes/CommissionedPaymentTest.java

```diff
@@ -0,0 +1,72 @@
+package com.mael.payroll.paymentTypes;
+
+import com.mael.payroll.cards.SalesCard;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.time.LocalDate;
+
+import static java.time.LocalDate.of;
+import static java.time.Month.JANUARY;
+import static org.junit.Assert.assertEquals;
+
+public class CommissionedPaymentTest {
+	private CommissionedPayment commissionedPayment;
+	private LocalDate fridayAndBiweekly;
+	private double amount;
+	private double commission;
+	private double monthlyRate;
+
+	@Before
+	public void setUp() {
+		monthlyRate         = 1000.0;
+		commission          = 2.0;
+		commissionedPayment = new CommissionedPayment(monthlyRate, commission);
+		fridayAndBiweekly   = of(2016, JANUARY, 29);
+		amount              = 200.0;
+	}
+
+	@Test
+	public void thePayIsZeroIfNoSalesCard() {
+		assertPay(0.0);
+	}
+
+	@Test
+	public void thePayIsRatePlusAmountByCommissionIfOneSalesCard() {
+		commissionedPayment.addSalesCard(new SalesCard(fridayAndBiweekly, amount));
+		assertPay(monthlyRate + (amount * commission / 100.0));
+	}
+
+	@Test
+	public void thePayIsTheSumOfRatesPlusSumOfCommissionsIfTwoSalesCards() {
+		LocalDate monday = of(2016, JANUARY, 25);
+		commissionedPayment.addSalesCard(new SalesCard(monday, amount));
+
+		LocalDate tuesday = of(2016, JANUARY, 26);
+		commissionedPayment.addSalesCard(new SalesCard(tuesday, amount));
+
+		assertPay(monthlyRate + (amount * commission / 100.0) +
+				  monthlyRate + (amount * commission / 100.0));
+	}
+
+	@Test
+	public void thePayIsCalculatedWithTheSalesCardsOfBiweeklyPeriodOnly() {
+		LocalDate fridayThreeWeeksAgo = of(2016, JANUARY, 8);
```

Comment:
Might be my head messing with me ;)
But a variable name like `currentWeek` implies to me that it depends on the actual _current_ week. Depending as in if I run the unit tests next week a different value will be used compared to when I run them today.

I think a good trade-off and good step forward would be to declare the pivot date (which the other dates rely upon) as the first variable, yes.